orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RestartStageHandler.kt

```diff
@@ -58,13 +59,34 @@ class RestartStageHandler(
         if (topStage.status.isComplete) {
           topStage.addRestartDetails(message.user)
           topStage.reset()
+          restartParentPipelineIfNeeded(message, topStage)
           repository.updateStatus(topStage.execution.type, topStage.execution.id, RUNNING)
           queue.push(StartStage(startMessage))
         }
       }
     }
   }
 
+  private fun restartParentPipelineIfNeeded(message: RestartStage, topStage: Stage) {
+    if (topStage.execution.trigger !is PipelineTrigger) {
+      return
+    }
+
+    val trigger = topStage.execution.trigger as PipelineTrigger
+
+    val parentStage = trigger.parentExecution.stageById(trigger.parentPipelineStageId)
+    parentStage.addSkipRestart()
+
```

Comment:
I believe you will need a `repository.storeStage(parentStage)` here to make sure your changes are persisted

Reply:
Shouldn't I use this instead?
```
repository.updateStageContext(parentStage)
```